File: python/Быстрый старт/5. 29 задача аналитики.py
"""
В трек аналитики второго сезона вошло 28 задач, однако кроме них была подготовлена ещё одна задача. Так как в сезон она не вошла, то предлагаем решить вам её сейчас.

Вам дано квадратное уравнение вида:
ax2+bx+c=0,ax 2
 +bx+c=0,
где a,ba,b и cc — целые числа, такие что a≠0a=0.

Напишите программу, которая решает это уравнение.
"""
import math


def main():
    temp = list(input().split(" "))
    a, b, c = float(temp[0]), float(temp[1]), float(temp[2])
    discr = b ** 2 - 4 * a * c

    if discr > 0:
        x1 = (-b + math.sqrt(discr)) / (2 * a)
        x2 = (-b - math.sqrt(discr)) / (2 * a)
        print("2")
        print("%.6f %.6f" % (x2, x1))
    elif discr == 0:
        x = -b / (2 * a)
        print("1")
        print("%.6f" % x)
    else:
        print("0")


if __name__ == '__main__':
    main()

File: golang/Быстрый старт/5. 29 задача аналитики.go
package main

import (
    "bufio"
    "os"
    "fmt"
    "math"
    "strconv"
    "strings"
)

func main() {
	myscanner := bufio.NewScanner(os.Stdin)
	myscanner.Scan()
	line := myscanner.Text()
	sli := strings.Fields(line)
	a, _ := strconv.ParseFloat(sli[0], 64)
	b, _ := strconv.ParseFloat(sli[1], 64)
	c, _ := strconv.ParseFloat(sli[2], 64)

	discr := math.Pow(b, 2) - 4*a*c
	if discr > 0 {
		x1 := (-b + math.Sqrt(discr)) / (2 * a)
		x2 := (-b - math.Sqrt(discr)) / (2 * a)
		fmt.Println("2")
		fmt.Println(fmt.Sprintf("%f %f", x2, x1))
	} else if discr == 0 {
		x := -b / (2 * a)
		fmt.Println("1")
		fmt.Println(fmt.Sprintf("%f", x))
	} else {
		fmt.Println("0")
	}

}
